taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -102,10 +120,30 @@ const ListTranslations = ({ nodeObject, setNodeObject }) => {
                 }
             }
         })
+
+        // add shownTranslationLanguages as a tag to otherLangTags
+        shownTranslationLanguages.forEach((languageCode) => {
+            if (!addedLanguageCodes.includes(languageCode)) {
+                otherLangTags.push({'languageCode' : languageCode, 'tags' : []})
+                addedLanguageCodes.push(languageCode)
+            }
+        })
+
         // Set states
+        setAddedLanguageCodes(addedLanguageCodes);
         setMainLangRenderedTranslations(mainLangTags);
         setRenderedTranslations(otherLangTags);
-    }, [nodeObject]);
+    }, [nodeObject, shownTranslationLanguages]);
+
+    useEffect(() => {
+        // get shown translation languages from local storage
+        try{
+            const shownTranslationLanguages = JSON.parse(localStorage.getItem(translationVisibilityPreferenceKey)) || []; 
+            setShownTranslationLanguages(shownTranslationLanguages);
```

Comment:
we may want to check if all are valid languages (as people can mess around with the local storage)